### PR TITLE
feat: setup animation framework for radial charts

### PR DESCRIPTION
**Pull request by rishabhmore**
# Description, Motivation and Context:
This PR aims to establish the same animation framework for animating any data updates for Pie Charts

### CODE REVIEW

File: lib/src/charts/widgets/bar_chart.dart [hunk 1]
```diff
@@ -27,6 +27,12 @@ class BarChart extends StatefulWidget {
   /// Defaults to true.
   final bool autoAnimate;
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/bar_chart.dart]: Let's keep the naming convention consistent. 
If the other param is called `animateOnLoad` let's call this `animateOnUpdate`

File: lib/src/charts/widgets/bar_chart.dart [hunk 2]
```diff
@@ -27,6 +27,12 @@ class BarChart extends StatefulWidget {
...
 
+  /// The Duration for which the chart should animate
+  final Duration? animDuration;
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/bar_chart.dart]: Use full names please. 
```suggestion
  final Duration? animationDuration;
```

File: lib/src/charts/widgets/bar_chart.dart [hunk 3]
```diff
@@ -27,6 +27,12 @@ class BarChart extends StatefulWidget {
...
+
+  /// A custom Animation controller which can be provided by the user
+  final AnimationController? animation;
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/bar_chart.dart]: `which can be provided by the user` makes this sound like this is internal documentation.
```suggestion
  /// A custom Animation controller to drive the animation
  final AnimationController? animation;
```

File: lib/src/charts/widgets/bar_chart.dart [hunk 4]
```diff
@@ -45,6 +51,8 @@ class BarChart extends StatefulWidget {
     this.chartHeight,
     this.animateOnLoad = true,
     this.autoAnimate = true,
+    this.animDuration,
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/bar_chart.dart]: Provide the default value here instead.

File: lib/src/charts/widgets/bar_chart.dart [hunk 5]
```diff
@@ -67,10 +73,11 @@ class _BarChartState extends State<BarChart> with TickerProviderStateMixin {
       targetData: [widget.data],
       animateOnLoad: widget.animateOnLoad,
       autoAnimate: widget.autoAnimate,
-      animation: AnimationController(
-        duration: const Duration(seconds: 1),
-        vsync: this,
-      ),
+      animation: widget.animation ??
+          AnimationController(
+            duration: widget.animDuration ?? const Duration(milliseconds: 500),
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/bar_chart.dart]: And remove from here
```suggestion
            duration: widget.animDuration,
```

File: lib/src/charts/widgets/pie_chart.dart [hunk 6]
```diff
@@ -20,6 +16,23 @@ class PieChart extends StatefulWidget {
...
+  /// Defaults to true.
+  final bool animateOnLoad;
+
+  /// Controls if the charts should auto animate any updates to the data
+  ///
+  /// Defaults to true.
+  final bool autoAnimate;
+
+  /// The Duration for which the chart should animate
+  final Duration? animDuration;
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/pie_chart.dart]: Same changes as BarChart

File: lib/src/charts/widgets/pie_chart.dart [hunk 7]
```diff
@@ -40,45 +57,40 @@ class PieChart extends StatefulWidget {
...
-      minValue: minValue,
-      maxValue: maxValue,
+    // provide the chart details to the controller
+    _controller = RadialController(
+      targetData: [widget.data],
+      animateOnLoad: widget.animateOnLoad,
+      autoAnimate: widget.autoAnimate,
+      animation: widget.animation ??
+          AnimationController(
+            duration: widget.animDuration ?? const Duration(milliseconds: 500),
```
Comment (shounak-mulay) [inline lib/src/charts/widgets/pie_chart.dart]: Set default value in constructor of widget.
Any time you can avoid null check and assign by providing a default value in the constructor itself and making the param non nullable, always prefer that

**Review by shounak-mulay — CHANGES_REQUESTED** [review]

File: lib/src/controllers/radial_controller.dart [hunk 8]
```diff
@@ -1,16 +1,51 @@
...
-  final double maxValue;
+class RadialController extends ChangeNotifier
+    with RadialDataMixin, ChartAnimationsMixin<RadialSeries> {
+  final Map<RadialSeries, RadialConfig> _seriesConfigs = {};
+  final Map<Type, RadialPainter> painters = {};
+
+  @override
+  double minValue = 0.0;
+  @override
+  double maxValue = 0.0;
```
Comment (shounak-mulay) [inline lib/src/controllers/radial_controller.dart:21]: Should we reset this values as well just like the other controller?

**Review by shounak-mulay — APPROVED** [review]

File: lib/src/controllers/radial_controller.dart (hunk 8 above)
Comment (rishabhmore) [reply lib/src/controllers/radial_controller.dart:21]: done

**Pull request merged**